c2rust-refactor/src/context.rs

```diff
@@ -689,23 +889,38 @@ impl<'hir> HirMap<'hir> {
     #[inline]
     pub fn opt_node_to_hir_id(&self, id: NodeId) -> Option<HirId> {
         if id > self.max_node_id {
-            None
-        } else {
-            if let Some(ldid) = self.node_id_to_def_id.get(&id) {
-                return Some(self.map.local_def_id_to_hir_id(*ldid));
-            }
+            return None;
+        }
+
+        if let Some(ldid) = self.node_id_to_def_id.get(&id) {
+            return Some(self.map.local_def_id_to_hir_id(*ldid));
+        }
 
-            self.ast_span_maps
-                .node_id_to_span_map
-                .get(&id)
-                .and_then(|span| self.span_to_hir_map.get(&span).copied())
+        if let Some(node_span) = self.ast_span_maps.node_id_to_span_map.get(&id) {
+            if let Some(context_key) = self.ast_span_maps.node_id_to_context_map.get(&id) {
+                if let Some(hir_id) = self
+                    .context_to_hir_map
+                    .get(&(*node_span, context_key.clone()))
+                {
+                    return Some(*hir_id);
+                }
+            }
+            return self.span_to_hir_map.get(node_span).copied();
```

Comment:
Is there a case where a span would be in the old map but not the new context one?

Reply:
There should always be at least one in the new context map, but there may be more since the span is ambiguous